shared/src/sources/balancer/pool_fetching.rs

```diff
@@ -105,19 +108,48 @@ impl BalancerPoolEvaluating for WeightedPool {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub struct AmplificationParameter {
+    factor: U256,
+    precision: U256,
+}
+
+impl AmplificationParameter {
+    pub fn new(factor: U256, precision: U256) -> Self {
+        Self { factor, precision }
+    }
+
+    /// This is the format used to pass into smart contracts.
+    pub fn as_u256(&self) -> U256 {
+        self.factor
+    }
+
+    /// This is the format used to pass along to HTTP solver.
+    pub fn as_big_rational(&self) -> Result<BigRational> {
+        // In order to circumvent the possibility of panic from BigRational::new,
+        // we check for zero denominator and return result
+        ensure!(!self.precision.is_zero(), "Zero precision not allowed");
+        Ok(BigRational::new(
+            self.factor.to_big_int(),
+            self.precision.to_big_int(),
+        ))
```

Comment:
I find it a bit silly/weird that we circumvent the panic, but then just use `.expect` on the result later...

Reply:
Included some suggestions :point_up: